core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -391,7 +391,12 @@ private static void setLocalDictInfo(CarbonTable table, TableInfo tableInfo) {
    */
   private void fillCreateOrderColumn(String tableName) {
     List<CarbonColumn> columns = new ArrayList<CarbonColumn>();
-    List<CarbonDimension> dimensions = this.tableDimensionsMap.get(tableName);
+    List<CarbonDimension> dimensions = new ArrayList<CarbonDimension>();
+    for (CarbonDimension dimension : this.tableDimensionsMap.get(tableName)) {
+      if (dimension.getColumnSchema().getSchemaOrdinal() != -1) {
```

Comment:
please add comment when getSchemaOrdinal = -1 and why skipping for it

Reply:
**I checked, in multiple places getSchemaOrdinal = -1 is already used, that means -1 has some meaning. Please have a separate flag to identify instead of this.**

Reply:
@VenuReddy2103 for index column please add one extra property which can be used to differentiate between table column and index column. Do not use -1

Reply:
Ok. Added a new property to identify it as an indexColumn.